orangecontrib/text/preprocess.py

```diff
@@ -1,94 +1,219 @@
     """
-    def __init__(self, incl_punct=False, lowercase=True, stop_words='english', trans=None, min_df=1):
+    def __init__(self, lowercase=True, stop_words=None, min_df=None, max_df=None, transformation=None,
+                 use_twitter_tokenizer=False):
         """
-        :param incl_punct: Determines whether the tokenizer should include punctuation in the tokens.
-        :type incl_punct: boolean
         :param lowercase: If set, transform the tokens to lower case, before returning them.
         :type lowercase: boolean
-        :param stop_words: Determines whether stop words should("english"), or should not(None) be removed.
-            If this is list, it should contain stopwords.
-        :type stop_words: 'english' or list or None
-        :param trans: An optional pre-processor object to perform the morphological
-            transformation on the tokens before returning them.
-        :type trans: :class: `orangecontrib.text.preprocess.Lemmatizer`
-            or :class: `orangecontrib.text.preprocess.Stemmer`
-        :return: :class: `orangecontrib.text.preprocess.Preprocessor`
-        """
-        # TODO Needs more elaborate check on list contents.
-        if not (stop_words == 'english' or isinstance(stop_words, list) or stop_words is None):
-            raise ValueError("The stop words parameter should be either \'english\', a list or None.")
-        if isinstance(trans, Stemmatizer):  # hack, since scikit does not obey lowercase when preprocesor given
-            trans.lowercase = lowercase
-        self.stop_words = stop_words
-        # TODO does including punctuation even work?
-        self.incl_punct = incl_punct
+        :param stop_words: Determines whether and what stop words should be removed. Can remove
+            default nltk stop words for the English language or stop words provided in a custom list.
+        :type stop_words: 'english', a list or None
+        :param min_df: Tokens that appear in exactly or less than 'min_df' documents, will be removed.
+            Can be specified either as an actual count of documents or a proportion of the corpus. Assigning
+            this parameter when pre-processing a single document, will have no effect.
+        :type min_df: int or float
+        :param max_df: Tokens that appear in exactly or more than 'max_df' documents, will be removed.
+            Can be specified either as an actual count of documents or a proportion of the corpus. Assigning
+            this parameter when pre-processing a single document, will have no effect.
+        :type max_df: int or float
+        :param transformation: Name of the morphological transformation method to be performed on the tokens.
+        :type transformation: `orangecontrib.text.preprocess.Stemmatizer`
+        :param use_twitter_tokenizer: Determines the use of either the Twitter or default word tokenizer.
+        :type use_twitter_tokenizer: bool
+        :return: list
```

Comment:
Would you consider [Napoleon-compatible docstrings](http://sphinxcontrib-napoleon.readthedocs.org/en/latest/index.html) format for new code?

That's `Settings | Tools | Python Integrated Tools | Docstring format` in PyCharm if you'd wish to switch to NumPy or Google.